apis/operator/v1alpha1/telemetry_types.go

```diff
@@ -38,8 +38,51 @@ const (
 
 // TelemetrySpec defines the desired state of Telemetry
 type TelemetrySpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Trace  TraceSpec  `json:"trace,omitempty"`
+	Metric MetricSpec `json:"metric,omitempty"`
+}
+
+// MetricSpec is the specification of the desired behavior of the metric gateway
+type MetricSpec struct {
+	Gateway MetricGatewaySpec `json:"gateway,omitempty"`
+}
+
+type MetricGatewaySpec struct {
+	Scaling Scaling `json:"scaling,omitempty"`
+}
+
+// TraceSpec is the specification of the desired behavior of the trace gateway
+type TraceSpec struct {
+	Gateway TraceGatewaySpec `json:"gateway,omitempty"`
+}
+
+type TraceGatewaySpec struct {
+	Scaling Scaling `json:"scaling,omitempty"`
+}
+
+// Scaling defines desired type scaling and number of desired pods.
```

Comment:
Do we even need the word "desired"?

```suggestion
// Scaling defines the scaling type and number of pods.
```